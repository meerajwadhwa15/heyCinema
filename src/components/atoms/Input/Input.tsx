import React, { FunctionComponent, SyntheticEvent } from 'react';

export type Props = {
  name: string;
  placeholder?: string;
  type?: string;
  value?: string;
  onChange?: (e: SyntheticEvent) => void;
};

const Input: FunctionComponent<Props> = props => <input {...props} />;

export default Input;
