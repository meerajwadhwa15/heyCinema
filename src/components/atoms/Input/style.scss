input {
  border: 1px solid $primaryBackgroundColor;
  font-style: italic;
}
