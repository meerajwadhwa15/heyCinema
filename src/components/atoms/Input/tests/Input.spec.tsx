import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import Input, { Props } from '../Input';

const props: Props = {
  name: 'search',
  placeholder: 'search',
  type: 'text',
  value: 'Test'
};

describe('Input Enzyme testing', () => {
  test('renders without crashing', () => {
    shallow(<Input {...props} />);
  });
});

describe('Input Snapshot testing', () => {
  test('Matches the snapshot', () => {
    const component = create(<Input {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
