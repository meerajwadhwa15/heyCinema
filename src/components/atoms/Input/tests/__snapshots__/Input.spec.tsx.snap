// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Snapshot testing Matches the snapshot 1`] = `
<input
  name="search"
  placeholder="search"
  type="text"
  value="Test"
/>
`;
